### PR TITLE
Fix subscribers merge issue

### PR DESCRIPTION
**Pull request by tabman83**
Fixes an issue while merging subscribers from KV and Cosmos where the DLQ subscriber is not recognized when already present in KV. The merger will now consider the SourceSubscriptionName in KV when present (and that's only there for DLQ subscribers) and match it with the SubscriptionName in Cosmos.

### CODE REVIEW

**Review by chomick — APPROVED** [review]
For future, the PR/branch name should include the ticket number (if not created, we should have one added).

**Review by mszymeshop — APPROVED** [review]

File: src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs [hunk 1]
```diff
@@ -161,9 +184,10 @@ public async Task WhenSameEventsExistInKvSubscribersAndCosmosSubscribers_CosmosS
 
             using (new AssertionScope())
             {
-                result.Data.Should().HaveCount(5);
+                result.Data.Should().HaveCount(6);
```
Comment (chomick) [inline src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs:187]: Do we have a test that merges both KV & Cosmos? This test, contrary to what the name states, is actually checking if unique KV & Cosmos events are maintained in the result, not that Cosmos wins? Or do I read it wrong?

File: src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs [hunk 2]
```diff
@@ -145,13 +162,19 @@ public async Task WhenSameEventsExistInKvSubscribersAndCosmosSubscribers_CosmosS
...
+                new SubscriberConfigurationBuilder().WithType("testEVENT").CreateAsDlq(),
+
                 new SubscriberConfigurationBuilder().WithType("testevent").WithSubscriberName("subscriber1").Create(),
                 new SubscriberConfigurationBuilder().WithType("testevent.completed").Create(),
             };
 
             var cosmosSubscribers = new[]
             {
-                new SubscriberBuilder().WithEvent("TESTevent").WithWebhook("https://cosmos.eshopworld.com/testevent/", "POST", "*", BasicAuthenticationEntity).Create(),
+                new SubscriberBuilder().WithEvent("TESTevent")
```
Comment (chomick) [inline src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs:173]: It looks the events are case sensitive (not ideal :( ) so all entries are unique

File: src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs (hunk 1 above)
Comment (tabman83) [reply src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs:187]: This is testing that Cosmos wins (and it's not case sensitive matching) as `TESTevent` in Cosmos will override `testEVENT` in KV

File: src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs (hunk 2 above)
Comment (tabman83) [reply src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs:173]: it is - the line above will override the kv model `testEVENT`

File: src/CaptainHook.DirectorService/Infrastructure/ConfigurationMerger.cs [hunk 3]
```diff
@@ -33,8 +33,10 @@ public ConfigurationMerger(ISubscriberEntityToConfigurationMapper subscriberEnti
         {
             var onlyInKv = subscribersFromKeyVault
                 .Where(kvSubscriber => !subscribersFromCosmos.Any(cosmosSubscriber =>
-                    kvSubscriber.EventType.Equals(cosmosSubscriber.ParentEvent.Name, StringComparison.InvariantCultureIgnoreCase)
-                    && kvSubscriber.SubscriberName.Equals(cosmosSubscriber.Name, StringComparison.InvariantCultureIgnoreCase)));
+                    kvSubscriber.EventType.Equals(cosmosSubscriber.ParentEvent.Name, StringComparison.InvariantCultureIgnoreCase) &&
```
[the remark this answers is not in the document]
Comment (tabman83) [reply src/CaptainHook.DirectorService/Infrastructure/ConfigurationMerger.cs:36]: added

**Review by mszymeshop — APPROVED** [review]

**Review by chomick — APPROVED** [review]

File: src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs (hunk 2 above)
Comment (chomick) [reply src/Tests/CaptainHook.Tests/Director/ConfigurationMergerTests.cs:173]: All good - missed the fact there are 8 subscribers added in total in the test, not 6.

**Pull request merged**